### PR TITLE
Event Code Access

### PR DESCRIPTION
**Pull request by RajeevReddyIlavala**

### CODE REVIEW

File: mercury/views/views.py [hunk 1]
```diff
@@ -9,3 +11,22 @@ def get(self, request, **kwargs):
 
 class AboutPageView(TemplateView):
     template_name = "about.html"
+
+
+class EventAccess(TemplateView):
+    def post(self, request, *args, **kwargs):
+        event_code = request.POST.get("eventcode")
+        event_access_object = EventCodeAccess.objects.latest("id")
```
Comment (de846) [inline mercury/views/views.py]: Does this code only check the "latest" event code in the table? What if an earlier event code needs to be re-enabled? Or, what if a future event code is put in the database but shouldn't be enabled yet? 

Have you considered doing a get/select using a filter to try and find a matching object that matches the event code, then checking if that match is enabled?

@MichaelLally  can you review the behavior?

File: mercury/views/views.py [hunk 2]
```diff
@@ -9,3 +11,22 @@ def get(self, request, **kwargs):
...
+        event_code_actual = event_access_object.event_code
+        if event_code == event_code_actual:
+            return HttpResponseRedirect("index")
+        else:
+            return HttpResponse("Invalid Event Code")
+
+    def get(self, request, **kwargs):
+        event_access_object = EventCodeAccess.objects.latest("id")
```
Comment (de846) [inline mercury/views/views.py]: I'm not sure I understand the behavior here. If someone tries to GET the login page, it checks the latest event code, and brings you to the "homepage" (index.html) if it is _not_ enabled? Can describe more the intention of the functionality?

File: mercury/views/views.py (hunk 1 above)
Comment (RajeevReddyIlavala) [reply mercury/views/views.py]: Well thought David, I considered that scenario where you can search for the event code provided by the user in the database but what if it’s a past event code that is not yet disabled by the admin. 
Here I am not sure of the requirements, so I am just picking the top row for testing  rest of the functionalities.
@MichaelLally please clarify on the desired behavior

**Comment by de846** [comment]
I also think that given the instructions from the professor and the TA, we need to insert a permanently available event_code when the server starts up since they have communicated that we need to be able to tolerate a "database reset" and still have our site functional.

File: mercury/views/views.py (hunk 1 above)
Comment (de846) [reply mercury/views/views.py]: If it is a "past event code" that is not disabled by the admin, then it is enabled, and should be viable. It is up to the admin to enable and disable event codes properly.

File: mercury/views/views.py (hunk 2 above)
Comment (RajeevReddyIlavala) [reply mercury/views/views.py]: - Disabled: app links directly to landing page
- Enabled: must enter event code to access landing page/website
The above is the description provided in the user story, I assumed that I achieved the functionality by redirecting the user to index.html bypassing the login page if event code is disabled.
I am not sure whether it's the right way to do it. Please let me know the right way to do it, if I am wrong here.

**Comment by RajeevReddyIlavala** [comment]
Will be replaced with another PR

**Pull request closed without merging**